interfaces/prompting/prompting.go

```diff
@@ -172,9 +171,6 @@ func (lifespan LifespanType) ValidateExpiration(expiration time.Time, currTime t
 		if expiration.IsZero() {
 			return prompting_errors.NewInvalidExpirationError(expiration, fmt.Sprintf("cannot have unspecified expiration when lifespan is %q", lifespan))
 		}
-		if currTime.After(expiration) {
-			return fmt.Errorf("%w: %q", prompting_errors.ErrRuleExpirationInThePast, expiration)
-		}
```

Comment:
Now that rules can have mixed outcomes/lifespans/expirations, it may be the case that (e.g. when reading from disk) one permission has expired and another one has not. Rather than throwing an error when an expired permission is seen, it instead is removed from the rule at a later step, and any non-expired permissions remain.

In particular: `ValidateExpiration` is only called once, from within `RulePermissionEntry.validate()`, which is in turn only called once, in `RulePermissionMap.validateForInterface()`. Before calling `entry.validate()`, `validateForInterface` first checks `entry.Expired()`, and if the permission entry has expired, the permission is removed from the permission map at the end of the function, after ensuring that no other errors occurred.

So not quite a simplification, it's a change in the distinction between expired rules and invalid rules now that rules can be "partially expired" but still valid.